usermods/Internal_Temperature_v2/usermod_internal_temperature.h

```diff
@@ -0,0 +1,134 @@
+#pragma once
+
+#include "wled.h"
+
+class InternalTemperatureUsermod : public Usermod
+{
+
+private:
+  unsigned long loopInterval;
+  unsigned long lastTime = 0;
+  bool isEnabled;
+  float temperature = 0;
+
+  static const char _name[];
+  static const char _enabled[];
+  static const char _loopInterval[];
+
+  // any private methods should go here (non-inline methosd should be defined out of class)
+  void publishMqtt(const char *state, bool retain = false); // example for publishing MQTT message
+
+public:
+  void setup()
+  {
+  }
+
+  void connected()
+  {
+  }
+
+  void loop()
+  {
+    // if usermod is disabled or called during strip updating just exit
+    // NOTE: on very long strips strip.isUpdating() may always return true so update accordingly
+    if (!isEnabled || strip.isUpdating() || millis() - lastTime <= loopInterval)
+      return;
+
+    lastTime = millis();
+
+#ifdef ESP8266 // ESP8266
+    // does not seem possible
+    temperature = -1;
+#elif defined(CONFIG_IDF_TARGET_ESP32S2) // ESP32S2
+    temperature = -1;
+#else                                    // ESP32 ESP32S3 and ESP32C3
```

Comment:
How accurate is the temperature function? Is it justified to have two decimal places, i.e. is the temperature value accurate up to 1/100th?

Reply:
Not so accurate actually. I think one decimal place will be enough.

Reply:
Changed to one decimal place